From 654a42d85bf4ed01628f53241cd28d55440fbc15 Mon Sep 17 00:00:00 2001
From: He Sheng <hesheng@wxiat.com>
Date: Tue, 7 Jun 2022 17:42:32 +0800
Subject: [PATCH] sw64: do some cleanups for rt_sigframe
Patch-mainline: NO, OTHERS
References: openEuler-22.03-LTS
Modified-by-SEL: No


Sunway inclusion
category: feature
bugzilla: https://gitee.com/openeuler/kernel/issues/I56OLG

--------------------------------

The retcode is unused, since we take __vdso_rt_sigreturn as default
restorer, and redundant err check in setup_rt_frame() can be removed.

Signed-off-by: He Sheng <hesheng@wxiat.com>

Signed-off-by: Gu Zitao <guzitao@wxiat.com>
Acked-by: Xie XiuQi <xiexiuqi@huawei.com>
Signed-off-by: Zheng Zengkai <zhengzengkai@huawei.com>
Signed-off-by: Guoqing Jiang <guoqing.jiang@suse.com>
---
 arch/sw_64/kernel/signal.c | 8 --------
 1 file changed, 8 deletions(-)

diff --git a/arch/sw_64/kernel/signal.c b/arch/sw_64/kernel/signal.c
index f7b8d391251a..17cc9850daaa 100644
--- a/arch/sw_64/kernel/signal.c
+++ b/arch/sw_64/kernel/signal.c
@@ -34,7 +34,6 @@ asmlinkage void ret_from_sys_call(void);
 struct rt_sigframe {
 	struct siginfo info;
 	struct ucontext uc;
-	unsigned int retcode[3];
 };
 
 /*
@@ -45,10 +44,6 @@ struct rt_sigframe {
 extern char compile_time_assert
 	[offsetof(struct rt_sigframe, uc.uc_mcontext) == 176 ? 1 : -1];
 
-#define INSN_MOV_R30_R16	0x47fe0410
-#define INSN_LDI_R0		0x201f0000
-#define INSN_CALLSYS		0x00000083
-
 static long
 restore_sigcontext(struct sigcontext __user *sc, struct pt_regs *regs)
 {
@@ -263,9 +258,6 @@ setup_rt_frame(struct ksignal *ksig, sigset_t *set, struct pt_regs *regs)
 	 */
 	r26 = VDSO_SYMBOL(current->mm->context.vdso, rt_sigreturn);
 
-	if (err)
-		return -EFAULT;
-
 	/* "Return" to the handler */
 	regs->r26 = r26;
 	regs->r27 = regs->pc = (unsigned long) ksig->ka.sa.sa_handler;
-- 
2.34.1

